Submodule manylinux contains modified content
diff --git a/manylinux/docker/build_scripts/build-cpython.sh b/manylinux/docker/build_scripts/build-cpython.sh
index a913760..be01e27 100755
--- a/manylinux/docker/build_scripts/build-cpython.sh
+++ b/manylinux/docker/build_scripts/build-cpython.sh
@@ -33,7 +33,7 @@ pushd Python-${CPYTHON_VERSION}
 PREFIX="/opt/_internal/cpython-${CPYTHON_VERSION}"
 mkdir -p ${PREFIX}/lib
 CFLAGS_EXTRA=""
-CONFIGURE_ARGS="--disable-shared --with-ensurepip=no"
+CONFIGURE_ARGS="--enable-shared --with-ensurepip=no"
 
 if [ "${2:-}" == "nogil" ]; then
 	PREFIX="${PREFIX}-nogil"
@@ -88,6 +88,8 @@ make install > /dev/null
 popd
 rm -rf Python-${CPYTHON_VERSION} Python-${CPYTHON_VERSION}.tgz Python-${CPYTHON_VERSION}.tgz.asc
 
+patchelf --force-rpath --set-rpath '$ORIGIN/../lib' ${PREFIX}/bin/python3
+
 if [ "${OPENSSL_PREFIX}" != "" ]; then
 	rm -rf ${OPENSSL_PREFIX}/bin ${OPENSSL_PREFIX}/include ${OPENSSL_PREFIX}/lib/pkgconfig ${OPENSSL_PREFIX}/lib/*.so
 fi
